xml/src/main/java/com/baeldung/xml/XMLDocumentWriter.java

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.xml;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+
+public class XMLDocumentWriter {
+    
+    public void write(Document document, String fileName, boolean excludeDeclaration, boolean prettyPrint) {
+        try {
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            if(excludeDeclaration) {
+                transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            }
+            if(prettyPrint) {
+                transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+                transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            }
+            DOMSource source = new DOMSource(document);
+            FileWriter writer = new FileWriter(new File(fileName));
+            StreamResult result = new StreamResult(writer);
+            transformer.transform(source, result);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Let's not encourage this practice. At least, let's do `throw new IllegalStateException(e)` or `throw new IllegalArgumentException(e)`, depending on whether we think it is a system error or user error.

Reply:
Updated code as per suggestion